sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -540,32 +540,45 @@ private static boolean isPortResource(Protos.Resource resource) {
         return getNewExecutorInfo(podInstance.getPod(), serviceName, targetConfigurationId);
     }
 
-    private static Protos.ContainerInfo getContainerInfo(ContainerSpec containerSpec) {
+    private static Protos.ContainerInfo getContainerInfo(ContainerSpec containerSpec,
+                                                         Collection<NetworkSpec> networks) {
         Protos.ContainerInfo.Builder containerInfo = Protos.ContainerInfo.newBuilder()
                 .setType(Protos.ContainerInfo.Type.MESOS);
 
-        if (containerSpec.getImageName().isPresent()) {
+        if (containerSpec != null && containerSpec.getImageName().isPresent()) {
             containerInfo.getMesosBuilder()
                     .setImage(Protos.Image.newBuilder()
                             .setType(Protos.Image.Type.DOCKER)
                             .setDocker(Protos.Image.Docker.newBuilder()
                                     .setName(containerSpec.getImageName().get())));
         }
 
-        if (!containerSpec.getNetworks().isEmpty()) {
+        if (networks.size() > 0) {
             containerInfo.addAllNetworkInfos(
-                    containerSpec.getNetworks().stream().map(n -> getNetworkInfo(n)).collect(Collectors.toList()));
+                    networks.stream().map(n -> getNetworkInfo(n)).collect(Collectors.toList()));
         }
 
-        if (!containerSpec.getRLimits().isEmpty()) {
+        if (containerSpec != null && !containerSpec.getRLimits().isEmpty()) {
             containerInfo.setRlimitInfo(getRLimitInfo(containerSpec.getRLimits()));
         }
 
         return containerInfo.build();
     }
 
     private static Protos.NetworkInfo getNetworkInfo(NetworkSpec networkSpec) {
-        return Protos.NetworkInfo.newBuilder().setName(networkSpec.getName()).build();
+        LOGGER.info("Loading NetworkInfo for network named \"" + networkSpec.getNetworkName() + "\"");
```

Comment:
slf4j's `Logger`, at least, supports bracket-style formatting like so: 
`LOGGER.info("Loading NetworkInfo for network named \"{}\"", networkSpec.getNetworkName())`
which is a little easier to read.